src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (i === 0) {
+      arr[i] = { ...state };
+    } else {
+      arr[i] = { ...arr[i - 1] };
+    }
```

Comment:
you don't need these, create an empty array outside the loop and push to this array result of each loop iteration